streaming/src/java/org/apache/hive/streaming/RecordWriter.java

```diff
@@ -69,9 +87,25 @@
   void close() throws StreamingException;
 
   /**
-   * Get the set of partitions that were added by the record writer.
+   * Get the set of partitions that were added were used but may have been
+   * added or not to the metastore.
    *
    * @return - set of partitions
    */
   Set<String> getPartitions();
+
+  /**
+   * Returns the location of the delta directory.
+   * @param partitionValues partition values
+   * @param bucketId bucket id
+   * @param minWriteId min write Id
+   * @param maxWriteId max write Id
+   * @param statementId statement Id
+   * @param table table
+   * @return the location of the file
+   * @throws StreamingException when the path is not found
+   */
+  Path getDeltaFileLocation(List<String> partitionValues,
```

Comment:
default? NiFi implements this interface afaik.. without default this may cause build failure for nifi.